### PR TITLE
feat: 🎸 Add confidential transaction related APIs

### PR DESCRIPTION
**Pull request by prashantasdeveloper**
### JIRA Link 

Adds APIs to create, affirm, execute a confidential transaction.
Adds APIs to get involved parties in a confidential transaction and to get confidential venues for a DID
Adds API to set venue filtering for a confidential asset
Adds API to verify sender proof as auditor/receiver
Adds API to decrypt balance for a confidential account

### Changelog / Description 

DA-1066, DA-1071, DA-1072, DA-1099, DA-1069, DA-1075, DA-1073, DA-1105, DA-1106, DA-1107

### Checklist - 

- [x] New Feature ?
- [x] Updated swagger annotation (if API structure is changed) ?
- [x] Unit Test (if possible) ?
- [x] Updated the Readme.md (if required) ?

### CODE REVIEW

**Review by polymath-eric — CHANGES_REQUESTED** [review]
Left some general comments, but after reading it all have a larger design comment.

I think it would be cleaner to make the proof module a self contained module with its own controller, and register it only if the config is provided.

That would clean up all of the "Note the setting must be configured", and will prevent any coupling with the other modules. The fact we need a side car service is a necessary evil for not, but we should try to quantine it in a self contained module.

File: src/confidential-transactions/confidential-transactions.service.ts [hunk 1]
```diff
@@ -46,4 +58,113 @@ export class ConfidentialTransactionsService {
...
+    );
+
+    const proofs = [];
+
+    for (const legAmount of legAmounts) {
+      const { amount, confidentialAsset } = legAmount;
+      const assetAuditor = assetAuditors.find(({ asset }) => asset.id === confidentialAsset);
+
+      if (!assetAuditor) {
+        throw new AppValidationError('Asset not found in the leg');
```
[the remark this answers is not in the document]
Comment (prashantasdeveloper) [reply src/confidential-transactions/confidential-transactions.service.ts:113]: Asset is missing here. This is the case when lets say you are adding a leg amount for an asset not present in the leg

File: src/proof-server/entities/confidential-account.entity.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+/* istanbul ignore file */
+
+export class ConfidentialAccountEntity {
```
[the remark this answers is not in the document]
Comment (prashantasdeveloper) [reply src/proof-server/entities/confidential-account.entity.ts:3]: I had in mind that anything not exposed outside of the nest wrapper would be an entity. We already use Model and Dto with output and input classes respectively. 

In this case, this is the object returned by the proof server's API. Labeling it as model will create confusion IMO with our standard model class

**Review by polymath-eric — APPROVED** [review]
lgtm, one nit on a doc comment

**Pull request merged**